stdlib/public/SDK/XCTest/XCTest.swift

```diff
@@ -146,17 +146,17 @@ public func XCTAssertNil(_ expression: @autoclosure () throws -> Any?, _ message
     if !passed {
       // TODO: @auto_string expression
       
-      _XCTRegisterFailure(true, _XCTFailureDescription(assertionType, 0, expressionValueStr as NSString), message, file, line)
+      _XCTRegisterFailure(true, _XCTFailureDescription(assertionType, 0, expressionValueStr as NSString), message(), file, line)
```

Comment:
Would it make sense to remove `@autoclosure` from the `message:` parameter of the `_XCTRegisterFailure` helper function, now? Are there any remaining use cases for that to be `@autoclosure`?

/cc @briancroom

Reply:
Considering that `_XCTRegisterFailure` unconditionally evaluates the `message` parameter, I doubt there's any need for it to actually be an autoclosure.

Reply:
Do you guys want me to make that change or would you handle it?

Reply:
FWIW: XCTest as well as the majority of other overlays is built in Swift 4 mode, and so this change is not strictly required. Thanks @xedin for fixing it anyways. LGTM, removing an autoclosure completely also works, and can probably be done separately.